source/operand.cpp

```diff
@@ -292,25 +296,29 @@ spv_operand_type_t spvTakeFirstMatchableOperand(
   assert(!pattern->empty());
   spv_operand_type_t result;
   do {
-    result = pattern->front();
-    pattern->pop_front();
+    result = pattern->back();
+    pattern->pop_back();
   } while (spvExpandOperandSequenceOnce(result, pattern));
   return result;
 }
 
 spv_operand_pattern_t spvAlternatePatternFollowingImmediate(
     const spv_operand_pattern_t& pattern) {
   spv_operand_pattern_t alternatePattern;
-  for (const auto& operand : pattern) {
-    if (operand == SPV_OPERAND_TYPE_RESULT_ID) {
-      alternatePattern.push_back(operand);
-      alternatePattern.push_back(SPV_OPERAND_TYPE_OPTIONAL_CIV);
-      return alternatePattern;
-    }
-    alternatePattern.push_back(SPV_OPERAND_TYPE_OPTIONAL_CIV);
+  // TODO: make this not terrible again: scan for RESULT_ID first,
+  // then decide what to do. (chrisforbes) not really bothered by
+  // this being a bit slow-- is a weird case in assembler path.
+
+  for (auto it = pattern.crbegin(); it != pattern.crend(); it++) {
+      if (*it == SPV_OPERAND_TYPE_RESULT_ID) {
+          alternatePattern.insert(alternatePattern.begin(), *it);
+          alternatePattern.insert(alternatePattern.begin(), SPV_OPERAND_TYPE_OPTIONAL_CIV);
+          return alternatePattern;
```

Comment:
Indeed it's a true corner case for making extra-weird test cases.  Don't bother making this fast.